src/Haskell/Ide/Engine/Plugin/HsImport.hs

```diff
@@ -79,25 +75,51 @@ importModule uri modName =
         Nothing -> do
           newText <- liftIO $ T.readFile output
           liftIO $ removeFile output
-          J.WorkspaceEdit mChanges mDocChanges <- liftToGhc $ makeDiffResult input newText fileMap
+          J.WorkspaceEdit mChanges mDocChanges <- liftToGhc
+            $ makeDiffResult input newText fileMap
 
           if shouldFormat
             then do
-              -- Format the import with Brittany
-              confFile <- liftIO $ Brittany.getConfFile origInput
-              newChanges <- forM mChanges $ mapM $ mapM (formatTextEdit confFile)
-              newDocChanges <- forM mDocChanges $ mapM $ \(J.TextDocumentEdit vDocId tes) -> do
-                ftes <- forM tes (formatTextEdit confFile)
-                return (J.TextDocumentEdit vDocId ftes)
-
-              return $ IdeResultOk (J.WorkspaceEdit newChanges newDocChanges)
-            else
+              config  <- getConfig
+              plugins <- getPlugins
+              let mprovider = Hie.getFormattingPlugin config plugins
+              case mprovider of
+                Nothing -> 
+                  return $ IdeResultOk (J.WorkspaceEdit mChanges mDocChanges)
+
+                Just (plugin, _) -> do
+                  newChanges <- forM mChanges $ \change -> do
+                    let func = mapM (formatTextEdit plugin)
+                    res <- mapM func change
+                    return $ fmap flatten res 
+
+                  newDocChanges <- forM mDocChanges $ \change -> do
+                    let cmd (J.TextDocumentEdit vids edits) = do
+                          newEdits <- mapM (formatTextEdit plugin) edits
+                          return $ J.TextDocumentEdit vids (flatten newEdits)
+                    mapM cmd change
+
+                  return $ IdeResultOk
+                    (J.WorkspaceEdit newChanges
+                                     newDocChanges
+                    )
+            else 
               return $ IdeResultOk (J.WorkspaceEdit mChanges mDocChanges)
 
-  where formatTextEdit confFile (J.TextEdit r t) = do
-          -- TODO: This tab size of 2 spaces should probably be taken from a config
-          ft <- fromRight t <$> liftIO (Brittany.runBrittany 2 confFile t)
-          return (J.TextEdit r ft)
+ where
+flatten :: List [a] -> List a
```

Comment:
Are these functions supposed to be inside the where? The indentation might be putting them at the top level

Reply:
Yes they are, maybe one of the formatters did something weird.